Stratis.Bitcoin/Features/Miner/PosMinting.cs

```diff
@@ -204,51 +226,78 @@ public void GenerateBlocks(WalletSecret walletSecret)
                         stakeTx.UtxoSet = set;
                         stakeTx.Secret = walletSecret; // Temporary.
                         stakeTxes.Add(stakeTx);
+                        this.logger.LogTrace("UTXO '{0}/{1}' with value {2} might be available for staking.", stakeTx.OutPoint.Hash, stakeTx.OutPoint.N, utxo.Value);
                     }
                 }
 
                 // Trying to sign a block.
                 if (this.SignBlock(stakeTxes, pblock, pindexPrev, pblocktemplate.TotalFee))
                 {
+                    this.logger.LogTrace("POS block signed successfully.");
                     var blockResult = new BlockResult { Block = pblock };
-                    this.CheckState(new ContextInformation(blockResult, this.network.Consensus), pindexPrev);
+                    this.CheckStake(new ContextInformation(blockResult, this.network.Consensus), pindexPrev);
 
                     pblocktemplate = null;
                 }
                 else
                 {
+                    this.logger.LogTrace("{0} failed, waiting {1} ms for next round...", nameof(this.SignBlock), this.minerSleep);
                     Task.Delay(TimeSpan.FromMilliseconds(this.minerSleep), this.nodeLifetime.ApplicationStopping).GetAwaiter().GetResult();
                 }
             }
         }
 
-        private void CheckState(ContextInformation context, ChainedBlock pindexPrev)
+        private void CheckStake(ContextInformation context, ChainedBlock pindexPrev)
         {
+            this.logger.LogTrace("({0}:'{1}')", nameof(pindexPrev), pindexPrev?.HashBlock);
+
             Block block = context.BlockResult.Block;
 
             if (!BlockStake.IsProofOfStake(block))
+            {
+                this.logger.LogTrace("(-)[NOT_POS]");
                 return;
+            }
 
             // Verify hash target and signature of coinstake tx.
             BlockStake prevBlockStake = this.stakeChain.Get(pindexPrev.HashBlock);
             if (prevBlockStake == null)
+            {
+                this.logger.LogTrace("(-)[NO_PREV_STAKE]");
                 ConsensusErrors.PrevStakeNull.Throw();
+            }
 
             context.SetStake();
             this.posConsensusValidator.StakeValidator.CheckProofOfStake(context, pindexPrev, prevBlockStake, block.Transactions[1], block.Header.Bits.ToCompact());
 
             // Found a solution.
             if (block.Header.HashPrevBlock != this.chain.Tip.HashBlock)
+            {
+                this.logger.LogTrace("(-)[REORG]");
```

Comment:
ah yes, there even is a comment, i will fix this in later PR, so pls ignore here, it will read

```
                this.logger.LogTrace("(-)[SOLUTION_FOUND]");

```